src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -644,86 +642,27 @@ public String getDisplayName() {
         }
 
         public FormValidation doCheckLogoutUrl(@QueryParameter String logoutUrl) {
-            if (StringUtils.isEmpty(logoutUrl)) {
-                return FormValidation.ok();
-            }
-            try {
-                new URL(logoutUrl);
-            } catch (MalformedURLException e) {
-                return FormValidation.error(ERROR_MALFORMED_URL, e);
-            }
-            return FormValidation.ok();
+            return SamlFormValidation.checkUrlFormat(logoutUrl);
         }
 
         public FormValidation doCheckDisplayNameAttributeName(@QueryParameter String displayNameAttributeName) {
-            if (StringUtils.isEmpty(displayNameAttributeName)) {
-                return FormValidation.ok();
-            }
-
-            if (StringUtils.isBlank(displayNameAttributeName)) {
-                return FormValidation.error(ERROR_ONLY_SPACES_FIELD_VALUE);
-            }
-
-            return FormValidation.ok();
+            return SamlFormValidation.checkStringFormat(displayNameAttributeName);
         }
 
         public FormValidation doCheckGroupsAttributeName(@QueryParameter String groupsAttributeName) {
-            if (StringUtils.isEmpty(groupsAttributeName)) {
-                return FormValidation.warning(WARN_RECOMMENDED_TO_SET_THE_GROUPS_ATTRIBUTE);
-            }
-
-            if (StringUtils.isBlank(groupsAttributeName)) {
-                return FormValidation.error(ERROR_ONLY_SPACES_FIELD_VALUE);
-            }
-
-            return FormValidation.ok();
+            SamlFormValidation.checkStringAttributeFormat(groupsAttributeName);
         }
 
         public FormValidation doCheckUsernameAttributeName(@QueryParameter String usernameAttributeName) {
-            if (StringUtils.isEmpty(usernameAttributeName)) {
-                return FormValidation.warning(WARN_RECOMMENDED_TO_SET_THE_USERNAME_ATTRIBUTE);
-            }
-
-            if (StringUtils.isBlank(usernameAttributeName)) {
-                return FormValidation.error(ERROR_ONLY_SPACES_FIELD_VALUE);
-            }
-
-            return FormValidation.ok();
+            return SamlFormValidation.checkStringAttributeFormat(usernameAttributeName);
```

Comment:
we'd lose the warning for the display name attribute, it is important to show these warning, this warning is only for this attribute not for every string attribute

Reply:
I only refactored the several usages in different less methods; which were 100% identically.